pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -285,7 +301,7 @@ func (hsm *hostStateMachine) handleProvisioningError(info *reconcileInfo) action
 }
 
 func (hsm *hostStateMachine) handleProvisioned(info *reconcileInfo) actionResult {
-	if hsm.Host.NeedsDeprovisioning() {
+	if hsm.provisioningCancelled() {
```

Comment:
Would a host that has already completed provisioning end up here?

Reply:
Yes. It's not a perfect name.